lib/code_corps_web/views/stripe_connect_account_view.ex

```diff
@@ -89,22 +89,47 @@ defmodule CodeCorpsWeb.StripeConnectAccountView do
 
   # recipient_status mapping
 
+  @doc ~S"""
+  Returns an inferred recipient verification status for the account, based on
+  the legal entity verification status and required fields for verification.
+
+  The default assumed status is "required".
+  If the verification status is "pending" and "legal_entity" fields are needed,
+  the returned status is "required".
+  If the veficication status
+  """
+  @spec recipient_status(StripeConnectAccount.t) :: String.t
   def recipient_status(stripe_connect_account, _conn) do
     get_recipient_status(stripe_connect_account)
   end
 
-  defp get_recipient_status(%StripeConnectAccount{legal_entity_verification_status: "unverified", verification_fields_needed: fields
-  }) do
-    cond do
-      Enum.member?(fields, "legal_entity.personal_id_number") -> "verifying"
-      Enum.member?(fields, "legal_entity.verification.document") -> "verifying"
+  @spec get_recipient_status(StripeConnectAccount.t) :: String.t
+  defp get_recipient_status(%StripeConnectAccount{
+    legal_entity_verification_status: "pending",
+    verification_fields_needed: needed_fields}) do
+
+    # https://stripe.com/docs/api#account_object-verification-fields_needed
+    # Check if the list of required fields includes any fields from the specified
+    # group.
+    # Required fields are listed as an array, nested in groups using `.`, example:
+    # `group_a.field_a`, `group_a.field_b`, `group_b.field_a`, etc.
+    case needed_fields |> includes_field_from?("legal_entity") do
```

Comment:
@joshsmith I rewrote this so I can clarify with a comment, but do keep in mind that **this change eliminates the `verifying status`** and I don't know what the reasoning behind it was.

If you did that by mistake and I reciprocated, we should revert the change.

Otherwise, we should document somewhere why we're doing it. I couldn't find any information about it.